src/main/java/org/apache/maven/plugins/assembly/internal/ComponentSupport.java

```diff
@@ -19,16 +19,18 @@
  * under the License.
  */
 
-import org.codehaus.plexus.archiver.zip.ZipUnArchiver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * @author jdcasey
- *
- *          //todo delete this class once the plexus maven plugin can merge a generated components.xml with an
- *          existing one.
+ * Support for components.
  */
-class SarUnArchiver
-    extends ZipUnArchiver
+public abstract class ComponentSupport
```

Comment:
I would deprecate this class right away if it will only contain logger injection.

Reply:
It is plugin internal anyway (look at pkg), is really just to lessen logger copy + pasta, but we can get rid of it, and just change all `getLogger()` occurences to `logger`

Reply:
I guess so, I'd go with `static final Logger LOGGER`.